pkg/digest/verification.go

```diff
@@ -113,22 +114,20 @@ func VerifyDir(root string, digest map[string]string) error {
 // not declare a signature path, use "".
 func ParseUris(
 	fetcher uri.Fetcher,
-	digestUri string,
-	signatureUri string,
+	digestUri *url.URL,
+	signatureUri *url.URL,
 ) (Digest, error) {
 	digest, err := fetcher.Open(digestUri)
 	if err != nil {
 		return Digest{}, err
 	}
 	defer digest.Close()
 	var signature io.ReadCloser
-	if signatureUri != "" {
-		signature, err = fetcher.Open(signatureUri)
-		if err != nil {
-			return Digest{}, err
-		}
-		defer signature.Close()
+	signature, err = fetcher.Open(signatureUri)
```

Comment:
Did you intent to make signatures mandatory?  If so, I'd like to see this in its own commit instead of hitching a ride with a commit that seems like a pure refactoring of types.